src/path.c

```diff
@@ -50,3 +51,70 @@ char *xcp_path_combine (const char *pathname, const char *subpath) {
 
   return res;
 }
+
+static inline const char *get_first_slash (const char *start, const char *pos) {
```

Comment:
You run the loop backward, from `pos - 1` to `start`, so doesn't it return the last slash position?

Reply:
The loop try to find the first slash before a given slash position until a non-slash char has been found.

Reply:
ok good to merge